server/routes/pt.js

```diff
@@ -114,4 +116,184 @@ router.post('/search', verifyToken, async (req, res) => {
   }
 });
 
+router.post('/addptrequest', verifyToken, async (req, res) => {
+  try {
+    const pt_id = req.body.pt_id; // ID of PT that was requested
+    const client_id = req.user.userId; // ID of client that made request
+ 
+    Pt.findByIdAndUpdate(pt_id,
+      {$push: {requests: client_id}},
+      {safe: true, upsert: true},
+      function(err,doc) {
+        if(err){
+          console.log(err);
+        }else{
+          return res.status(OK).send(client_id);
+        }
+      });
+  }
+  catch (err) {
+    return res.status(BAD_REQUEST).send(err.message);
+  }
+  
+});
+
+router.post('/addclientrequest', verifyToken, async (req, res) => {
+  try {
+    const pt_id = req.body.pt_id; // ID of PT that was requested
+    const client_id = req.user.userId; // ID of client that made request
+    
+    Client.findByIdAndUpdate(client_id,
```

Comment:
This needs to be done at the same time as `/addptrequest`. Merge these two routes.
Also you should probably use `findById()` instead, like this:

```js
const pt = await Pt.findById(pt_id);
const client = await Client.findById(client_id);
if (!pt || !client) return res.status(NOT_FOUND).send();
pt.requests.push(client_id);
client.requests.push(pt_id);
pt.save();
client.save();
```

Reply:
these two routes are being called at the same time, I only separated them for readability